src/node_buffer.cc

```diff
@@ -730,12 +730,20 @@ void ReadDoubleBE(const FunctionCallbackInfo<Value>& args) {
 
 
 template <typename T, enum Endianness endianness>
-uint32_t WriteFloatGeneric(const FunctionCallbackInfo<Value>& args) {
+void WriteFloatGeneric(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  bool noAssert = args[3]->BooleanValue();
+  if (!noAssert) {
```

Comment:
I believe the condition should be `args[3]->IsUndefined() || !args[3]->BooleanValue()`. Alternatively, we could use `!!` on the JS side to ensure we only get booleans at this point the same way the other arguments are type coerced `value = +value; offset = offset >>> 0;`. Do you have a preference between these?